### PR TITLE
Listening task

### PR DESCRIPTION
**Pull request by KristinaSig**
Person| Total Word Count | Section(s)
-- | -- | --
Frederik | 631 words | Implementation
Kristina | 811 words | Motivation + Materials
Elena | 601 words | Task Flow

### CODE REVIEW

File: _posts/2021-03-01-listening.md [hunk 1]
```diff
@@ -8,4 +8,104 @@ sections:
...
+
+
+<div id="motivation"></div>
+## Motivation
+
+The first three tasks have been designed to provide practice with core language skills concerned with literacy – the ability to read, write and construct sentences in a foreign language. However, as Renukadevi (2014) suggests, complete language proficiency depends crucially on another skill, namely listening. Being able to comprehend spoken words is necessary for face-to-face communication, which is the most common end goal of studying a foreign language (eg. Morley, 2001) . Not only is it necessary in order to come up with an appropriate response, but importantly it also helps students to acquire the correct pronunciation, word stress and a more confident use of vocabulary and syntax, which all improve the students’ speaking competencies. Yildirim and Yildirim (2016) as well as others have pointed out that ever since school curriculums have transferred to a more communicative approach of foreign language teaching, listening skills have also been gaining in popularity in school environments. For this reason, we decided to create an additional task that can provide users with practice and more exposure to spoken English.
```
Comment (polina-tsvilodub) [inline _posts/2021-03-01-listening.md:23]:
```suggestion
The first three tasks have been designed to provide practice with core language skills concerned with literacy – the ability to read, write and construct sentences in a foreign language. However, as Renukadevi (2014) suggests, complete language proficiency depends crucially on another skill, namely listening. Being able to comprehend spoken words is necessary for face-to-face communication, which is the most common end goal of studying a foreign language (e. g., Morley, 2001). Not only is it necessary in order to come up with an appropriate response, but, importantly, it also helps students to acquire the correct pronunciation, word stress and a more confident use of vocabulary and syntax, which all improve the students’ speaking competencies. Yildirim and Yildirim (2016) as well as others have pointed out that ever since school curricula have transferred to a more communicative approach of foreign language teaching, listening skills have also been gaining in popularity in school environments. For this reason, we decided to create an additional task that can provide users with practice and more exposure to spoken English.
```

File: _posts/2021-03-01-listening.md [hunk 2]
```diff
@@ -8,4 +8,104 @@ sections:
...
+
+<div id="materials"></div>
+## Materials
+
+At the moment, the audio database contains 45 recordings, each centered around a certain topic and labelled according to difficulty level. All files come from publicly available websites which are intended for learners and teachers of English as a foreign language, namely www.listenaminute.com, www.elllo.org and www.esl-lounge.com. We made sure that all sources legally allow the use of their audio materials for non-commercial purposes.
```
Comment (polina-tsvilodub) [inline _posts/2021-03-01-listening.md:43]:
```suggestion
At the moment, the audio database contains 45 recordings, each centered around a certain topic and labelled according to a difficulty level. All files come from publicly available websites which are intended for learners and teachers of English as a foreign language, namely [www.listenaminute.com](www.listenaminute.com), [www.elllo.org](www.elllo.org) and [www.esl-lounge.com](www.esl-lounge.com). We made sure that all sources legally allow the use of their audio materials for non-commercial purposes.
```

File: _posts/2021-03-01-listening.md (hunk 1 above)
Comment (polina-tsvilodub) [reply _posts/2021-03-01-listening.md:23]: Sorry about the incorrect review formatting. I just fixed minor typos :)

File: _posts/2021-03-01-listening.md [hunk 3]
```diff
@@ -8,4 +8,104 @@ sections:
...
+ 
+In order to provide a more authentic learning experience, we aimed to provide a good variety of speakers with regard to their voices and accents, as well as a wide range of everyday topics, for which intermediate learners are expected to have the required core vocabulary. Some examples include Health, Media, French Fries and Job Duties. Unfamiliar accents can severely hinder students’ abilities to understand the spoken words (Gilakjani & Sabouri, 2016), so only speakers of the most common native English accents are included in the final data set.
+ 
+Since the maximum number of players is 4, we have designed 4 questions per audio, each with 3 possible options to choose from where exactly one is correct. The focus of the questions is also diverse. Some require a general understanding of the speaker’s attitude about the topic, others ask for a more detailed-oriented listening or distinguishing between homonyms (equally or similarly sounding words) based on context.
```
Comment (polina-tsvilodub) [inline _posts/2021-03-01-listening.md:51]:
```suggestion
Since the maximum number of players is four, we have designed four questions per audio, each with three possible options to choose from where exactly one is correct. The focus of the questions is also diverse. Some require a general understanding of the speaker’s attitude about the topic, others ask for a more detail-oriented listening or distinguishing between homonyms (equally or similarly sounding words) based on context.
```

File: _posts/2021-03-01-listening.md [hunk 4]
```diff
@@ -8,4 +8,104 @@ sections:
...
+ 
+In terms of the difficulty level, the main motivation for the division was to eventually accommodate the adaptive module, which assigns a specific question to the users based on their previous performance. There does not seem to be a unified official method for determining the difficulty level of a listening task. Therefore, our process was as follows: the content (in form of a textual transcript) was first evaluated based on the language complexity and vocabulary using a combination of automatic tools ([1], [2]) to indicate a CEFR level [3] of a text. In addition to this indication, the data was also subjectively evaluated for speech qualities such as tempo and clarity, and the final level of difficulty resulted from agreement between three judges.
```
Comment (polina-tsvilodub) [inline _posts/2021-03-01-listening.md:53]:
```suggestion
In terms of the difficulty level, the main motivation for the division was to eventually accommodate the adaptive module, which assigns a specific question to the users based on their previous performance. There does not seem to be a unified official method for determining the difficulty level of a listening task. Therefore, our process was as follows: the content (in form of a textual transcript) was first evaluated based on the language complexity and vocabulary using a combination of automatic tools ([1](http://www.roadtogrammar.com/textanalysis/), [2](https://cefr.duolingo.com)) to indicate a CEFR level [3] of a text. In addition to this indication, the data was also subjectively evaluated for speech qualities such as tempo and clarity, and the final level of difficulty resulted from agreement between three judges.
```

File: _posts/2021-03-01-listening.md [hunk 5]
```diff
@@ -8,4 +8,104 @@ sections:
...
+ 
+Although our target audience is mainly B1-B2 students, since the intermediate proficiency group tends to be quite heterogeneous in terms of the individuals’ strengths and weaknesses, we also included slightly simpler audios that could theoretically be used with motivated A2 students, too.
+ 
+Each recording was carefully reviewed by at least three people, checking for the sound quality of the recording, correctness of the language and multiple choice questions, flagging of potentially sensitive subjects or unsuitable language, and as mentioned, the indicated level of difficulty.
```
Comment (polina-tsvilodub) [inline _posts/2021-03-01-listening.md:57]:
```suggestion
Each recording was carefully reviewed by at least three people, checking for the sound quality of the recording, correctness of the language and multiple choice questions, flagging potentially sensitive subjects or unsuitable language, and as mentioned, checking the indicated level of difficulty.
```

**Review by polina-tsvilodub — COMMENTED** [review]
I think the Motivation & Materials sections are in a great shape, I only commented on minor typos. It provides a good overview and is written very concisely. My only optional question would be if there is place where potentially interested readers have the option to look at all the finalized materials. If so, maybe it could be linked to in the materials section. 

Great job!

**Comment by TobiObeck** [comment]
I read the whole page and made some comments. I saw that Emma and Polina already gave feedback and suggestions, so it might be possible that I accidentally mention something again. Please ignore in such a case.

**motivation**
- First paragraph very good. Nice, that you have so many sources. Just being curious, did you research all these sources yourself?

**Comment by TobiObeck** [comment]
**task flow**
- task flow. Long sentence. reasoning why it is at the very end doesn't seem sound to me. It kind sounds like: it is at the end therefore it is voluntary. Why not make the task an option in one of the first three stages of the game? Is it too hard, takes too much time or sth like that?
- How long are the audio clips (on average)? Is saw that this is explained further below in the materials section. Probably wouldn't hurt to have this information twice. It is debatable though.
- Why the decision to first play the audio clip, then present the questions? Why not vice versa? Ok this was explained later in the text. Maybe you could give a short note that the reasoning for this will be explained further below
- "After the selected player stated their answer, [THE] next question is being shown in the chat. [THE] same procedure applies to all questions" 
Insert the 'the's. Otherwise it sound too colloquial. 
- "In total, each text is assigned four different questions,..." shouldn't it be 'audio clip' instead of 'text'?
- "However, because of the Telegram chosen as a platform for the Escapeling, we have encountered a series of difficulties while designing this task." Sounds a bit weird. Also; I wouldn't make a line break after the sentence, since the sentence belongs to the paragraph below and shouldn't be alone.
How about: "Because Telegram was chosen as the platform for the application, we have ecnountered a series of difficulties while designing the task. First of all, ..."
- What is played twice? questions or audio?
- "In case of the Telegram" better "Because Telegram is used..."
Also very long sentence. "...get the audio file deleted from the chat [depending on a condition. For example the audio file should be deleted after a certain amount...]"

**Comment by TobiObeck** [comment]
**materials**
- At the moment, the audio database contains 45 recordings, each centered around a certain topic and labelled according to [a] difficulty level.
- "...even if the users decide to listen more times, ..." better "multiple times" 
- "Some require a general understanding of the speaker’s attitude about the topic, others ask for a more detailed-oriented listening or" better "more detailed" or "more detail-oriented"

**Comment by TobiObeck** [comment]
**technical architecture**
- First off all, the three images are not rendered on the website properly (I have a local build process running). I also don't know where they could be existing already. Therefore, my review & comments might be a bit more negative than they would be with the images, because it is harder for me to understand without them. Could you upload the files here: https://github.com/ALLUOS/ALLUOS.github.io/tree/master/assets/images
- "sheath" is a bit of an odd word choice. Maybe better "wrap"?
- capitalization: Lots of upper case words that should be lower case: "Option", "User", "Bot", "Answer", "Points", "Topic", "Button", "Users", "Seconds", "Exploitation"
- markdown formatting for code with backticks, not double quotes
- huge paragraph. I have a hard time understanding the details. Splitting it up into 3 paragraphs with one main idea for each paragraph would make the text easier to understand. Also this is a lot of stuff to keep in short-term memory. I feel like there is missing a guided thread or giving the reader some kind of structure what to expect. Instead there are a lot of different aspects of the implementation mentioned that make it kind of hard to follow through. I would imagine it would be beneficial if reference figure 1 repeatedly in your text instead of just mentioning it at the very end. Guide the reader through the diagram instead of letting him figure it out himself. 
- "was not able to change the state for technical reasons" Maybe that could be an especially interesting point, what exact technical difficulties you had and why it was not possible to change the state.
- "2/3rds" -> "two-thirds"

**Comment by TobiObeck** [comment]
**inline-keyboards**
- CallbackQueryHandler not formatted
- "The receiver of the event..." Sentence construction seems a bit confusing. Maybe sth like: "When this button is clicked an event is fired. This event is received by the `CallbackQueryHandler` which in turn calls the `time_is_up()` function." But why mention this detail in the first place? This feels like the beginning of bigger explanation but abruptly stops when figure two is introduced.

**access to the audio-files**
- Honestly, I feel like this section isn't really worth mentioning. Maybe it could be cut out entirely. Maybe this also gives you more words to describe the technical architecture more clearly.
- "To reduce the coding-Work a decision was made to collect the different topics in a json-file." coding-work sounds odd. Also isn't the reasoning rather to have all user-facing texts in one place and also the possibility to internationalize/translate it into different languages without the need to search through mutliple placed in the code?
- Two times "which". Better replace with sth. like: "The file consisted of several topics, which were structured as shown in the following figure"

**possible exploitations**
- This section is written clearly and reads very well! The only minor thing you could do is to split up the text in to two paragraphs. So that "These two problems could not be solved completely..." is the start of the second pararagraph.

**Review by karinakhokhlova — CHANGES_REQUESTED** [review]
Thank you for a very well described task flow of the new listening task. I especially liked the mention of the limitations and the comparison of a listening task in a real classroom vs a virtual one. I left some notes on a general grammatical mistakes. But overall this part looks pretty well written!

**Comment by TobiObeck** [comment]
Since I already have a Jekyll build process installed and configured on my machine to generate the web pages, I made you a screenshot and pdf so that you can check images and formatting. The pdf doesn't look 100% accurate but is maybe more convenient for reading than the huge image.

[listening_2021_08_28_13h30.pdf](https://github.com/ALLUOS/ALLUOS.github.io/files/7070647/listening_2021_08_28_13h30.pdf)

https://user-images.githubusercontent.com/13554426/131216731-0c4defa2-862c-45f6-936d-a8bb25e67b24.png

**Review by EmmaZanoli — APPROVED** [review]
